src/test/java/org/springframework/integration/kafka/listener/AbstractBrokerTest.java

```diff
@@ -0,0 +1,126 @@
+		MutableMultimap<Integer, Integer> partitionDistribution = createPartitionDistribution(partitionCount, brokers, replication);
+		AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(getKafkaRule().getZookeeperClient(), topicName, toKafkaPartitionMap(partitionDistribution), AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK$default$4(), AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK$default$5());
+		for (int i = 0; i < partitionDistribution.keysView().size(); i++) {
+			TestUtils.waitUntilMetadataIsPropagated(asScalaBuffer(getKafkaRule().getKafkaServers()), TEST_TOPIC, i, 5000L);
+		}
+	}
+
+	public MutableMultimap<Integer, Integer> createPartitionDistribution(int partitionCount, int brokers, int replication) {
+		MutableMultimap<Integer, Integer> partitionDistribution = Multimaps.mutable.list.with();
+		for (int i = 0; i < partitionCount; i++) {
+			for (int j = 0; j < replication; j++) {
+				partitionDistribution.put(i, (i + j) % brokers);
+			}
+		}
+		return partitionDistribution;
+	}
+
+
+	public Configuration getKafkaConfiguration() {
+		return new BrokerAddressListConfiguration(getKafkaRule().getBrokerAddresses());
+	}
+
+	public static scala.collection.Seq<KeyedMessage<String, String>> createMessages(int count) {
+		return createMessagesInRange(0,count-1);
+	}
+
+	public static scala.collection.Seq<KeyedMessage<String, String>> createMessagesInRange(int start, int end) {
+		List<KeyedMessage<String,String>> messages = new ArrayList<KeyedMessage<String, String>>();
+		for (int i=start; i<= end; i++) {
+			messages.add(new KeyedMessage<String, String>(TEST_TOPIC, "Key " + i, i, "Message " + i));
+		}
+		return asScalaBuffer(messages).toSeq();
+	}
+
+	public Producer<String, String> createStringProducer(int compression) {
+		Properties producerConfig = TestUtils.getProducerConfig(getKafkaRule().getBrokersAsString(), TestPartitioner.class.getCanonicalName());
+		producerConfig.put("serializer.class", StringEncoder.class.getCanonicalName());
+		producerConfig.put("key.serializer.class",  StringEncoder.class.getCanonicalName());
+		producerConfig.put("compression.codec",  Integer.toString(compression));
+		return new Producer<String, String>(new ProducerConfig(producerConfig));
+	}
+
+	public ConnectionFactory getKafkaBrokerConnectionFactory() throws Exception {
+		DefaultConnectionFactory connectionFactory = new DefaultConnectionFactory(getKafkaConfiguration());
+		connectionFactory.afterPropertiesSet();
+		return connectionFactory;
+	}
+
+	private Map toKafkaPartitionMap(Multimap<Integer, Integer> partitions) {
+		java.util.Map<Object, Seq<Object>> m = partitions.toMap().collect(new Function2<Integer, RichIterable<Integer>, Pair<Object, Seq<Object>>>() {
+			@Override
+			public Pair<Object, Seq<Object>> value(Integer argument1, RichIterable<Integer> argument2) {
+				return Tuples.pair((Object) argument1, List$.MODULE$.fromArray(argument2.toArray(new Object[0])).toSeq());
```

Comment:
`fromArray` is `deprecated`, so, mark this method with `@SuppressWarnings("deprecation")`